### PR TITLE
[TEVA-1331] Create and validate ACM certs with Terraform

### PR DESCRIPTION
**Pull request by nickromneydfe**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-1331

## Changes in this PR:

- Add creation and validation of Amazon Certificate Manager certificates via Terraform

## Next steps:

- [x] Merge PR 2062 first
- [x] Create new certificate via Terraform
- [ ] Update cloudfront certificate arn in secrets for staging, production env vars
- [ ] Remove existing certificate (expiring 2020-11-16T12:00:00UTC)

### CODE REVIEW

**Review by alexbowen — APPROVED** [review]
im not particuarly au fait with terraform configs but...

File: terraform/common/modules/domains/data.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+data aws_route53_zone selected {
+  # AWS treats *.domain.name and domain.name as the same validation record
```
Comment (saliceti) [inline terraform/common/modules/domains/data.tf:2]: Is this comment still useful?
Also, `salected` was approriate when we were filtering. Now we could have a more generic name like `all` or `zones`

File: terraform/common/modules/domains/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+output "arn" {
```
Comment (saliceti) [inline terraform/common/modules/domains/outputs.tf:1]: Extra double quotes

File: terraform/common/modules/domains/provider.tf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+provider "aws" {
```
Comment (saliceti) [inline terraform/common/modules/domains/provider.tf:1]: Extra double quotes

**Review by saliceti — APPROVED** [review]

File: terraform/common/modules/domains/data.tf (hunk 1 above)
Comment (nickromneydfe) [reply terraform/common/modules/domains/data.tf]: Removed comment
Changed from `selected` to `zone`

File: terraform/common/modules/domains/provider.tf (hunk 3 above)
Comment (nickromneydfe) [reply terraform/common/modules/domains/provider.tf]: Removed

**Pull request merged**